data/api_v1_data_helper.go

```diff
@@ -112,9 +113,45 @@ func (a *API) getLatestPumpSettings(ctx context.Context, traceID string, userID
 		a.logger.Printf("{%s} - {GetDiabeloopParametersHistory:\"%s\"}", traceID, err)
 	}
 	timeEnd(ctx, "getParamHistory")
+
+	timeIt(ctx, "getLatestBasalSecurityProfile")
+	lastestProfile, err := a.store.GetLatestBasalSecurityProfile(ctx, traceID, userID)
+	if err != nil {
+		writer.basalSecurityProfile = nil
+		a.logger.Printf("{%s} - {GetLatestBasalSecurityProfile:\"%s\"}", traceID, err)
+	}
+	writer.basalSecurityProfile = TransformToExposedModel(lastestProfile)
+	timeEnd(ctx, "getLatestBasalSecurityProfile")
+
 	return iterPumpSettings, nil
 }
 
+func TransformToExposedModel(lastestProfile *store.DbProfile) *internalSchema.Profile {
+	var result *internalSchema.Profile
+
+	if lastestProfile != nil {
+		result = &internalSchema.Profile{} 
+		// Build start and end schedule
+		for i, value := range lastestProfile.BasalSchedule {
+			var elem internalSchema.Schedule
+			elem.Rate = value.Rate
+			elem.Start = value.Start
+			if i == len(lastestProfile.BasalSchedule)-1 {
+				elem.End = 0
```

Comment:
lastestProfile.BasalSchedule[0].Start ? (or it is always 0 ? )

Reply:
it is always zero but indeed it is a good protection, will do it